corehq/apps/users/util.py

```diff
@@ -487,3 +481,18 @@ def bulk_auto_deactivate_commcare_users(user_ids, domain):
 
 def is_dimagi_email(email):
     return email.endswith('@dimagi.com')
+
+
+def is_username_available(username):
+    """
+    Checks if the username is available to use
+    :param username: expects complete/email formatted username (e.g., user@example.commcarehq.org)
+    :return: boolean
+    """
+    from corehq.apps.users.dbaccessors import user_exists
+    # enforce complete username, otherwise check is meaningless
+    if not re.match("[a-z0-9.+-_]+@[a-z0-9.+-_]+.commcarehq.org", username):
```

Comment:
The regex to match valid email addresses is quite a bit more complicated than this. What do you think of using Django's email validation function instead?
```suggestion
    from django.core.validators import validate_email
    # enforce complete username, otherwise check is meaningless
    if validate_email(username) and username.endswith(".commcarehq.org"):
```